omnikinverter/omnikinverter.py

```diff
@@ -102,22 +103,19 @@ async def request(
                 msg = "Error occurred while communicating with Omnik Inverter device"
                 raise OmnikInverterConnectionError(msg) from exception
 
-            types = ["application/json", "application/x-javascript", "text/html"]
-            content_type = response.headers.get("Content-Type", "")
-            if not any(item in content_type for item in types):
-                text = await response.text()
-                msg = "Unexpected response from the Omnik Inverter device"
-                raise OmnikInverterError(
-                    msg,
-                    {"Content-Type": content_type, "response": text},
-                )
-
             raw_response = await response.read()
         finally:
-            if self.session and self._close_session:
-                await self.session.close()
-                self.session = None
-                self._close_session = False
+            await self.close()
+
+        types = ["application/json", "application/x-javascript", "text/html"]
+        content_type = response.headers.get("Content-Type", "")
+        if content_type.split(";")[0] not in types:
```

Comment:
Do we have an example of a request that returns more than 1 content_type in the headers? Since otherwise I do not see the need to make this adjustment now.

Reply:
https://www.w3.org/Protocols/rfc1341/4_Content-Type.html according to the possible values for `Content-Type` it's always this fixed pattern of `type "/" subtype *[";" parameter]` (I think the `*` only allows multiple subtypes??). There's never more than one `Content-Type` and that's not allowed by this newer parser either: we're specifically looking for either the full string being one of the MIME types in the `types` array, or having some extra parameters after `;` which we cut off.

Alternatively this could have been an `if not any(content_type.starts_with(type) for type in types)` but that would allow for possible future mime-types with a common prefix to what's in the `types` array.

Reply:
The main point here is that I didn't agree with the original loose `if not any(item in content_type for item in types)` which allows the MIME type to show up _anywhere_ in `content_type` (also possibly allowing for newer MIME types with additional prefix/suffix).

Reply:
We could add some extra tests to validate that this works as intended, FWIW.

Even then I don't know if additional parameters (`;` and extras) are ever passed for these MIME types, so this should have just been `if not content_type in types`, but I figured the additional `item in content_type` is there for a reason so this _must_ have been the case. Let's bisect to be sure.

Reply:
`blame` says this was added when `application/json` support was added in https://github.com/klaasnicolaas/python-omnikinverter/commit/0e3473e8a3114704d30e7bdba96897ce614334bf#diff-521682543174b52631cd8f53b94cd68e51f00db32ac413a2a04651e41afc94ebR94. Perhaps the inverter replies with `; charset=utf8` there?

Reply:
I ask for an example, you don't give it to me except for a bucket of information that I didn't actually ask for.

Reply:
Okay then, I'll boil this down...

Consider an imaginary new `application/json-compressed` MIME type. `"application/json" in "application/json-compressed"` happily returns `True` even though we don't know/understand/support that type.

Again: this is not fixing any out-in-the-wild issues, just solidifying the existing checks a little bit. Otherwise we might as well remove this error as it's not very strict at all.